presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2213,7 +2215,10 @@ public ConnectorPartitioningHandle getPartitioningHandleForExchange(ConnectorSes
         return new HivePartitioningHandle(
                 partitionCount,
                 partitionTypes.stream()
-                        .map(type -> toHiveType(typeTranslator, translateHiveUnsupportedTypeForTemporaryTable(type, typeManager)))
+                        .map(type -> toHiveType(
```

Comment:
Let this thing fail for now. Ideally the `HivePartitioningHandle ` shouldn't contain the hive types if the partition handle is for materialized exchange.